cg/apps/demultiplex/sample_sheet/read_sample_sheet.py

```diff
@@ -50,6 +57,20 @@ def get_sample_sheet_from_file(
     )
 
 
+def get_sample_type_from_sequencer_type(sequencer_type: str) -> Type[FlowCellSample]:
+    bcl_converter: str = get_bcl_converter_by_sequencer(sequencer_type)
+    if bcl_converter == BclConverter.BCL2FASTQ:
+        return FlowCellSampleBcl2Fastq
+    return FlowCellSampleBCLConvert
+
+
+def get_bcl_converter_by_sequencer(sequencer_type: str) -> str:
+    """Return the BCL converter based on the sequencer."""
+    if sequencer_type in [Sequencers.NOVASEQ, Sequencers.NOVASEQX]:
+        return BclConverter.DRAGEN
+    return BclConverter.BCL2FASTQ
+
```

Comment:
Not a fan of determining the sample sheet sample type like this, ideas are welcome 😅

Reply:
If this is the sole purpose, then I guess you could skip the middle step and just return FlowCellSampleBcl2FastQ if sequencer_type in [Sequencers.NOVASEQ, Sequencers.NOVASEQX] else FlowCellSampleBCLConvert?

Maybe not making it much clearer but I am uncertain if the middle step helps.

Reply:
But in general I guess it should be possible from the sample sheet.

Reply:
Yes, I think so too - would be nice to maybe encapsulate it in the `get_sample_sheet_from_file` function

Reply:
I'll keep the current logic as is and open a separate PR to refactor it (maybe just avoiding the duplication or moving the type determination). I don't think this PR should make any changes to the logic used in the post processing flow